crates/steel/src/lib.rs

```diff
@@ -263,10 +269,20 @@ impl<D, F: EvmFactory> EvmEnv<D, F, Commitment> {
     }
 }
 
-/// An EVM abstraction of a block header.
+/// Steel abstraction of the EVM specification identifier (e.g., London, Shanghai).
+pub trait EvmSpecId: Ord {
+    /// Whether EIP-4788 has been activated.
+    fn has_eip4788(&self) -> bool;
+    /// Whether EIP-2935 has been activated.
+    fn has_eip2935(&self) -> bool;
+    /// Converts the specification ID into an `u32`. This is used to compute [ChainSpec::digest()].
+    fn to_u32(&self) -> u32;
```

Comment:
This is 100% compatible to the existing digest() computation. However, it could lead to clashes for different chains, but then the chainID would be different, so I don't this is a security issue.